src/layer/WmtsLayer.ts

```diff
@@ -0,0 +1,104 @@
+import axios, { AxiosRequestConfig } from 'axios';
+
+import { GetMapParams, ApiType } from './const';
+import { AbstractLayer } from './AbstractLayer';
+import { RequestConfiguration } from '../utils/cancelRequests';
+import { ensureTimeout } from '../utils/ensureTimeout';
+import { CACHE_CONFIG_30MIN } from '../utils/cacheHandlers';
+import { getAxiosReqParams } from '../utils/cancelRequests';
+import { bboxToXyz, fetchLayersFromWmtsGetCapabilitiesXml } from './wmts.utils';
+
+interface ConstructorParameters {
+  baseUrl?: string;
+  layerId?: string;
+  title?: string | null;
+  description?: string | null;
+  legendUrl?: string | null;
+  resourceUrl?: string | null;
+}
+
+export class WmtsLayer extends AbstractLayer {
+  protected baseUrl: string;
+  protected layerId: string;
+  protected resourceUrl: string;
+
+  public constructor({
+    baseUrl,
+    layerId,
+    title = null,
+    description = null,
+    legendUrl = null,
+    resourceUrl = null,
+  }: ConstructorParameters) {
+    super({ title, description, legendUrl });
+    this.baseUrl = baseUrl;
+    this.layerId = layerId;
+    this.resourceUrl = resourceUrl;
+  }
+
+  public async updateLayerFromServiceIfNeeded(reqConfig?: RequestConfiguration): Promise<void> {
+    await ensureTimeout(async innerReqConfig => {
+      if (!this.resourceUrl) {
+        const parsedLayers = await fetchLayersFromWmtsGetCapabilitiesXml(this.baseUrl, innerReqConfig);
+        const layer = parsedLayers.find(layerInfo => this.layerId === layerInfo.Name[0]);
+        this.resourceUrl = layer.ResourceUrl;
+      }
+    }, reqConfig);
+  }
+
+  public async getMap(params: GetMapParams, api: ApiType, reqConfig?: RequestConfiguration): Promise<Blob> {
+    return await ensureTimeout(async innerReqConfig => {
+      await this.updateLayerFromServiceIfNeeded(reqConfig);
+      const paramsWithoutEffects = { ...params };
```

Comment:
Why are we removing effects and never applying them?

Reply:
Thanks, didnt notice I wasnt adding the effects function